gstreamer-plugin/git-patch/0001-svthevcenc-Add-svthevc-encoder-element.patch

```diff
@@ -0,0 +1,2059 @@
++
++  g_object_class_install_property (gobject_class, PROP_QP,
++      g_param_spec_uint ("qp", "QP",
++          "QP value for intra frames",
++          0, 51, PROP_QP_DEFAULT, G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
++
++  g_object_class_install_property (gobject_class, PROP_QP_MAX,
++      g_param_spec_uint ("qp-max", "QP Max",
++          "Maximum QP value allowed for rate control use",
++          0, 51, PROP_QP_MAX_DEFAULT,
++          G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
++
++  g_object_class_install_property (gobject_class, PROP_QP_MIN,
++      g_param_spec_uint ("qp-min", "QP Min",
++          "Minimum QP value allowed for rate control use",
++          0, 50, PROP_QP_MIN_DEFAULT,
++          G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
++
++  g_object_class_install_property (gobject_class, PROP_SCENE_CHANGE_DETECTION,
++      g_param_spec_boolean ("sc-detection", "Scene Change Detection",
++          "Use the scene change detection algorithm",
++          PROP_SCENE_CHANGE_DETECTION_DEFAULT,
++          G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
++
++  g_object_class_install_property (gobject_class, PROP_TUNE,
++      g_param_spec_uint ("tune", "Tune",
++          "Quality tuning mode: 0=SQ(visually optimized mode), 1=OQ(PSNR/SSIM optimized mode), 2=VMAF(VMAF optimized mode)",
++          0, 2, PROP_TUNE_DEFAULT, G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
++
++  g_object_class_install_property (gobject_class, PROP_LATENCY_MODE,
++      g_param_spec_uint ("latency-mode", "Latency Mode",
++          "Flag to enable for lower latency mode: 0=Normal Latency, 1=Low Latency",
++          0, 1, PROP_LATENCY_MODE_DEFAULT,
++          G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
++
++  g_object_class_install_property (gobject_class, PROP_BASE_LAYER_SWITCH_MODE,
++      g_param_spec_uint ("baselayer-mode", "Base Layer Switch Mode",
++          "Random Access Prediction Structure type setting: 0=Use B-frames in the base layer pointing to the same past picture, 1=Use P-frames in the base layerRefer to Appendix A.1",
++          0, 1, PROP_BASE_LAYER_SWITCH_MODE_DEFAULT,
++          G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
++
++  g_object_class_install_property (gobject_class, PROP_BITRATE,
++      g_param_spec_uint ("bitrate", "Bitrate",
++          "Bitrate in kbit/sec",
++          0, G_MAXINT, PROP_BITRATE_DEFAULT,
++          G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
++
++  g_object_class_install_property (gobject_class, PROP_KEY_INT_MAX,
++      g_param_spec_int ("key-int-max", "Max key frame",
++          "Distance Between Intra Frame inserted: -1=no intra update. -2=auto",
++          -2, 255, PROP_KEY_INT_MAX_DEFAULT, G_PARAM_READWRITE));
++
++  g_object_class_install_property (gobject_class, PROP_INTRA_REFRESH,
++      g_param_spec_uint ("intra-refresh", "Intra Refresh",
++          "Use Periodic Intra Refresh instead of IDR frames: 1=CRA (Open GOP), 2=IDR (Closed GOP)",
```

Comment:
Remove "Use Periodic Intra Refresh instead of IDR frames: ", since in x264, it's a boolean.
"1=CRA (Open GOP), 2=IDR (Closed GOP)" should be fine.